lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -770,7 +769,30 @@ def _request(
             headers["Content-Type"] = "application/json"
 
         response = self._request_raw(method, path, query, headers, data)
-        return json.loads(response.read().decode())["result"]
+        response = json.loads(response.read().decode())
+        if response["type"] == "async":
+            return self._wait(response["change"])
+        return response["result"]
+
+    def _wait(self, change_id: int) -> JSONType:
```

Comment:
Let's have a maximum timeout here, maybe 300s (5 minutes).

Reply:
Also, it looks like `response["change"]` is actually a string (it only happens to be an integer inside the string). See [here](https://github.com/canonical/snapd/blob/4dd5dedffe85b9de33ca537a482b1787601ec262/daemon/api_snap_conf.go#L172) and [here](https://github.com/canonical/snapd/blob/4dd5dedffe85b9de33ca537a482b1787601ec262/overlord/state/change.go#L253).

Reply:
Added timeout and changed the typing for `change_id`, thanks!